pkg/ipamd/ipamd.go

```diff
@@ -1003,13 +1028,17 @@ func (c *IPAMContext) eniIPPoolReconcile(ipPool []string, attachedENI awsutils.E
 				log.Debugf("Reconcile skipping IP %s on ENI %s because it was recently unassigned from the ENI.", strPrivateIPv4, eni)
 				continue
 			} else {
-				log.Debugf("This IP was recently freed, but is out of cooldown. We need to verify with EC2 control plane.")
-				// Call EC2 to verify IPs on this ENI
-				ec2Addresses, err := c.awsClient.GetIPv4sFromEC2(eni)
-				if err != nil {
-					log.Error("Failed to fetch ENI IP addresses!")
-					continue
-				} else {
+				if isMetadataIPs {
+					// IMDS data might be stale
+					log.Debugf("This IP was recently freed, but is out of cooldown. We need to verify with EC2 control plane.")
+					// Call EC2 to verify IPs on this ENI
+					ec2Addresses, err := c.awsClient.GetIPv4sFromEC2(eni)
```

Comment:
Why are we doing one ec2 api call per IP? why cant we do one call per ENI and then validate all IPs?

Reply:
This only happens on IPs that are just out of cooldown. I think we should simplify this loop more to make this clear, and avoid calling EC2 more than once. That can be a follow up PR, or this change will get too big.

Reply:
Yes my concern was towards freed IPs. If I scale up to 1000 and then bring down to 1 then there were will be quite a few API calls. It’s just moving the ec2 api call outside the for loop isn’t?